pages/k8s/ipv6.md

```diff
@@ -0,0 +1,182 @@
+
+## Enabling IPv6 and dual-stack in Charmed Kubernetes
+
+These features can be used simply by changing the configuration for Calico and the 
+Kubernetes master to include the relevant CIDRs.
+
+For Calico, the `cidr` configuration can contain two comma-separated values, with the
+first CIDR in the list being the preferred family for pods.
+
+For the Kubernetes master, the `service-cidr` configuration can contain two 
+comma-separated values, with the first being the default family for services.
+
+Note that Kubernetes supports [explicitly setting the `IPFamily`][ip-family] for a
+service when creating it.
+
+The following example shows how to deploy Charmed Kubernetes with IPv6 and dual-stack
+enabled.
+
+
+### Example deployment
+
+You can use the following overlay file ([download it here][asset-ipv4-ipv6-overlay])
+along with [the Calico overlay][asset-calico-overlay] to enable IPv4-preferred
+dual-stack:
+
+```yaml
+description: Charmed Kubernetes overlay to enable IPv4-IPv6 dual-stack.
+applications:
+  calico:
+    options:
+      cidr: "192.168.0.0/16,fd00:c00b:1::/112"
+  kubernetes-master:
+    options:
+      service-cidr: "10.152.183.0/24,fd00:c00b:2::/112"
+```
+
+You can deploy all of that with:
+
+```
+juju deploy cs:charmed-kubernetes --overlay ./calico-overlay.yaml --overlay ipv4-ipv6-overlay.yaml
+```
+
+Once that is deployed, you can use the following spec ([download it
+here][asset-nginx-dual-stack]) to run a dual-stack enabled nginx pod with an
+IPv6 service in front of it:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginxdualstack
+spec:
+  selector:
+    matchLabels:
+      run: nginxdualstack
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: nginxdualstack
+    spec:
+      containers:
+      - name: nginxdualstack
+        image: rocks.canonical.com/cdk/diverdane/nginxdualstack:1.0.0
+        ports:
+        - containerPort: 80
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx6
+  labels:
+    run: nginxdualstack
+spec:
+  type: NodePort
+  ipFamily: IPv6
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: nginxdualstack
+```
+
+<a id='known-issues> </a>
+## Known Issues
+
+Because of the pre-release feature status for IPv6 and dual-stack in Kubernetes
+and since Juju does not officially support IPv6, there are currently
+some things which won't work 100% and some limitations on IPv6 configuration. 
+These will also vary depending on the underlying cloud provider.
+
+### Juju
+
+The following arise because Juju does not fully support IPv6:
+
+* The charms require IPv4 on the underlying hosts, even when running a cluster
+  in IPv6-only mode.
+
+* By default, connections to the API server will use the IPv4 address even when
+  running a cluster in IPv6-preferred or IPv6-only mode. This can be modified
+  in the client config by hand or overridden via the `loadbalancer-ips` config
+  on the [kubernetes-master] and / or kubeapi-load-balancer charms.
```

Comment:
Is this supposed to be a link?  Should they both be links to the charms, or perhaps the charm reference page?

Reply:
sorry, yes, my mistake - I was in the middle of creating links to the charm docs configure sections, but then I wondered about whether it could be enabled post install